docs: Small clarity change for except*

gstein opened the pull request:
In the `except*` discussion, four keywords are linked in a row: `try`, `break`, `continue`, and `return`. Visually, it is easy to miss the period after the `try`. Adding a couple words (and the capitalization) after the period helps the reader notice the separate concerns of the keywords.


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--121073.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

gstein committed:
Small clarity change for except*

In the except* discussion, four keywords are linked in a row: try, break, continue, and return. Visually, it is easy to miss the period after the "try". Adding a couple words (and the capitalization) after the period helps the reader notice the separate concerns of the keywords.

eendebakpt commented on Doc/reference/compound_stmts.rst:
```diff
@@ -383,6 +383,7 @@ have ambiguous semantics.
 
 It is not possible to mix :keyword:`except` and :keyword:`!except*`
 in the same :keyword:`try`.
+The statements
```
```suggestion
The keywords
```
Is a bit more specific.

But either way, I agree adding the words makes the text easier to read.

gstein replied:
While yes, they are keywords, the core issue demanding they cannot be used is because those three are _flow-control **statements**_ that would mess up processing of multiple `except*` clauses.

gstein replied:
In that sense, adding the two extra words ("flow-control"; hyphenated?) might provide even more clarity.

StanFromIreland reviewed (commented):
I don’t see a particular benefit, it reads fine to me either way. cc @iritkatriel

AA-Turner commented on Doc/reference/compound_stmts.rst:
```diff
@@ -383,6 +383,7 @@ have ambiguous semantics.
 It is not possible to mix :keyword:`except` and :keyword:`!except*`
 in the same :keyword:`try`.
+The statements
 :keyword:`break`, :keyword:`continue` and :keyword:`return`
```
```suggestion
The :keyword:`break`, :keyword:`continue`, and :keyword:`return` statements
```

AA-Turner commented on Doc/reference/compound_stmts.rst:
```diff
@@ -383,6 +383,7 @@ have ambiguous semantics.
 
 It is not possible to mix :keyword:`except` and :keyword:`!except*`
 in the same :keyword:`try`.
+The statements
```
```suggestion
```

AA-Turner reviewed (approved):
Looks fine, but shall we combine the lines:

gstein committed:
Combine lines.

I originally inserted two words for minimal change. The suggestion reads much better.

Co-authored-by: Adam Turner <9087854+AA-Turner@users.noreply.github.com>

gstein committed:
Combine lines, part 2.

Co-authored-by: Adam Turner <9087854+AA-Turner@users.noreply.github.com>

AA-Turner committed:
Merge branch 'main' into patch-1
